mods/taskbar-thumbnail-hover-fix.wh.cpp

```diff
@@ -0,0 +1,218 @@
+    inputs[1].mi.dwFlags = MOUSEEVENTF_MOVE;
+    inputs[1].mi.dy = 1;
+
+    SendInput(2, inputs, sizeof(INPUT));
+}
+
+bool IsCursorOverThumbnail() {
+    POINT pt;
+    if (!GetCursorPos(&pt)) {
+        return false;
+    }
+
+    HWND hwnd = WindowFromPoint(pt);
+    if (!hwnd) {
+        return false;
+    }
+
+    HWND root = GetAncestor(hwnd, GA_ROOT);
+    if (!root) {
+        root = hwnd;
+    }
+
+    WCHAR className[256];
+    if (GetClassName(root, className, ARRAYSIZE(className))) {
+        if (wcsstr(className, L"Xaml") != nullptr) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void CALLBACK WinEventProc(
+    HWINEVENTHOOK hWinEventHook,
+    DWORD event,
+    HWND hwnd,
+    LONG idObject,
+    LONG idChild,
+    DWORD idEventThread,
+    DWORD dwmsEventTime
+) {
+    if (idObject != OBJID_WINDOW || idChild != 0) {
+        return;
+    }
+
+    if (IsCursorOverThumbnail()) {
+        TriggerHoverRefresh();
+    }
+}
+
+bool IsMutexOwned() {
+    HANDLE testMutex = OpenMutexW(SYNCHRONIZE, FALSE, MUTEX_NAME);
+    if (testMutex) {
+        CloseHandle(testMutex);
+        return true;
+    }
+    return false;
+}
+
+DWORD WINAPI WorkerThreadProc(LPVOID lpParam) {
+    DWORD ourPid = GetCurrentProcessId();
+
+    // Wait for taskbar to appear and verify ownership
+    while (g_running) {
+        if (IsMutexOwned()) {
+            Wh_Log(L"Another instance already active");
+            return 0;
+        }
+
+        HWND taskbarWnd = FindWindowW(L"Shell_TrayWnd", nullptr);
+        if (taskbarWnd) {
+            DWORD taskbarPid = 0;
+            GetWindowThreadProcessId(taskbarWnd, &taskbarPid);
+
+            if (taskbarPid != ourPid) {
+                Wh_Log(L"Taskbar owned by different process");
+                return 0;
+            }
+
+            break;
+        }
+
+        Sleep(500);
+    }
+
+    if (!g_running) {
+        return 0;
+    }
+
+    // Acquire mutex
+    g_mutex = CreateMutexW(nullptr, TRUE, MUTEX_NAME);
+    if (!g_mutex) {
+        Wh_Log(L"Failed to create mutex: %u", GetLastError());
+        return 0;
+    }
+    if (GetLastError() == ERROR_ALREADY_EXISTS) {
+        CloseHandle(g_mutex);
+        g_mutex = nullptr;
+        Wh_Log(L"Another instance claimed mutex first");
+        return 0;
+    }
+
+    // Install WinEvent hook
+    g_winEventHook = SetWinEventHook(
+        EVENT_OBJECT_DESTROY, EVENT_OBJECT_DESTROY,
+        nullptr,
+        WinEventProc,
+        0, 0,
+        WINEVENT_OUTOFCONTEXT
+    );
+
+    if (!g_winEventHook) {
+        Wh_Log(L"Failed to install WinEventHook: %u", GetLastError());
+        ReleaseMutex(g_mutex);
+        CloseHandle(g_mutex);
+        g_mutex = nullptr;
+        return 1;
+    }
+
+    Wh_Log(L"Hook installed successfully");
+
+    // Message pump for WinEventHook
+    MSG msg;
+    while (g_running) {
+        DWORD result = MsgWaitForMultipleObjects(0, nullptr, FALSE, 100, QS_ALLINPUT);

```

Comment:
Instead of polling each 100ms, you can post a message to the thread to exit. Example:

https://github.com/ramensoftware/windhawk-mods/blob/7a97b948c1747274d849d2c53f6f88be9920f95f/mods/taskbar-background-helper.wh.cpp#L667-L682

https://github.com/ramensoftware/windhawk-mods/blob/7a97b948c1747274d849d2c53f6f88be9920f95f/mods/taskbar-background-helper.wh.cpp#L874-L879